### PR TITLE
[10.0][FIX] bi_view_editor: bugfixing

### PR DESCRIPTION
**Pull request by astirpe**
This PR does the following:

- [x] Avoid occasional error while deleting Custom BI Views: fixes https://github.com/OCA/reporting-engine/issues/158
- [x] Does not crash when selection fields are defined by function
- [x] Does not stuck when user double-clicks
- [x] Fixes the deletion of the items from the widget (only the last one could be deleted)
- [x] Adds js robustness
- [x] Fix some pylint warnings

### CODE REVIEW

**Comment by astirpe** [comment]
@jbeficent @lreficent you may be interested in this.

File: bi_view_editor/models/bve_view.py [hunk 1]
```diff
@@ -399,6 +402,9 @@ def action_reset(self):
                 self.action_id.view_id.sudo().unlink()
             self.action_id.sudo().unlink()
 
+        views = self.env['ir.ui.view'].sudo().search(
+            [('model', '=', self.model_name)])
+        views.sudo().unlink()
```
Comment (pedrobaeza) [inline bi_view_editor/models/bve_view.py]: No need of other sudo.

**Review by pedrobaeza — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1:  Thanks @astirpe

**Comment by astirpe** [comment]
@pedrobaeza is it ok to merge this one?

**Pull request merged**